File: core/src/net/brenig/pixelescape/screen/ui/general/HorizontalSpacer.java
package net.brenig.pixelescape.screen.ui.general;

import com.badlogic.gdx.scenes.scene2d.ui.Widget;

/**
 * Simple Horizontal Spacer
 */
public class HorizontalSpacer extends Widget {

	private final float minWidth;
	private final float prefWidth;
	private final float maxWidth;

	public HorizontalSpacer() {
		this(0, Float.MAX_VALUE, Float.MAX_VALUE);
	}

	public HorizontalSpacer(float minWidth) {
		this(minWidth, Float.MAX_VALUE, Float.MAX_VALUE);
	}

	public HorizontalSpacer(float minWidth, float prefWidth, float maxWidth) {
		this.minWidth = minWidth;
		this.prefWidth = prefWidth;
		this.maxWidth = maxWidth;
	}

	@Override
	public float getMinWidth() {
		return minWidth;
	}

	@Override
	public float getMaxWidth() {
		return maxWidth;
	}

	@Override
	public float getPrefWidth() {
		return prefWidth;
	}
}

File: core/src/net/brenig/pixelescape/render/ui/general/HorizontalSpacer.kt
package net.brenig.pixelescape.render.ui.general

import com.badlogic.gdx.scenes.scene2d.ui.Widget

/**
 * Simple Horizontal Spacer
 */
class HorizontalSpacer constructor(private val minWidth: Float = 0f, private val prefWidth: Float = Float.MAX_VALUE, private val maxWidth: Float = Float.MAX_VALUE) : Widget() {

    override fun getMinWidth(): Float {
        return minWidth
    }

    override fun getMaxWidth(): Float {
        return maxWidth
    }

    override fun getPrefWidth(): Float {
        return prefWidth
    }
}
